core/src/main/java/org/apache/gravitino/authorization/OwnershipManager.java

```
@@ -0,0 +1,138 @@
+  private static final Logger LOG = LoggerFactory.getLogger(OwnershipManager.class);
+  private final EntityStore store;
+
+  public OwnershipManager(EntityStore store) {
+    this.store = store;
+  }
+
+  public void setOwner(
+      NameIdentifier identifier, Entity.EntityType type, String ownerName, Owner.Type ownerType) {
+    try {
+      String metalake;
+      if (type == Entity.EntityType.METALAKE) {
+        metalake = identifier.name();
+      } else {
+        metalake = identifier.namespace().level(0);
+      }
+
+      if (ownerType == Owner.Type.USER) {
+        store
+            .relationOperations()
+            .insertRelation(
+                Relation.Type.OWNER_REL,
+                identifier,
+                type,
+                AuthorizationUtils.ofUser(metalake, ownerName),
+                Entity.EntityType.USER,
+                true);
+      } else if (ownerType == Owner.Type.GROUP) {
+        store
+            .relationOperations()
+            .insertRelation(
+                Relation.Type.OWNER_REL,
+                identifier,
+                type,
+                AuthorizationUtils.ofGroup(metalake, ownerName),
+                Entity.EntityType.GROUP,
+                true);
+      }
+    } catch (NoSuchEntityException nse) {
+      LOG.warn("Entity {} or owner {} is not found", identifier, ownerName, nse);
+      throw new NotFoundException(nse, "Entity %s or owner %s is not found", identifier, ownerName);
+    } catch (IOException ioe) {
+      LOG.info("Fail to set the owner {} of entity {}", ownerName, identifier, ioe);
+      throw new RuntimeException(ioe);
+    }
+  }
+
+  public Owner getOwner(NameIdentifier identifier, Entity.EntityType type) {
+    try {
+      OwnerImpl owner = new OwnerImpl();
+      List entities =
+          store
+              .relationOperations()
+              .listEntitiesByRelation(Relation.Type.OWNER_REL, identifier, type);
+      if (entities.isEmpty()) {
```

I'm afriad that existing entities may has no ower and then throw an exception here.

---

Yes, it will throw an exception. We should return the exception to the client, too.